api/v1beta1/vmagent_types.go

```diff
@@ -237,6 +237,9 @@ type VMAgentSpec struct {
 	// ExtraEnvs that will be added to VMAgent pod
 	// +optional
 	ExtraEnvs []v1.EnvVar `json:"extraEnvs,omitempty"`
+	// ServiceSpec that will be added to vmagent service spec
+	// +optional
+	ServiceSpec v1.ServiceSpec `json:"serviceSpec,omitempty"`
```

Comment:
Can you add custom type for serviceSpec:

```go
 type ServiceSpec struct {
 Name string
 Spec: v1.ServiceSpec
  Annotations: map[string]string
  Labels: map[string]string
```

 and use it as 
``` go
ServiceSpec *ServiceSpec
```
 for all data structures.

Reply:
The struct looks more reasonable, I will fix it, thanks.